src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -515,8 +535,13 @@ public void Dispose()
         {
             // Let current batch saving task finish.
             this.cancellation.Cancel();
+
+            if (this.saveAsyncLoopException == null)
```

Comment:
Is this needed?

Reply:
ah the task will throw so you prevent it from throwing.
There is a race here if we shutdown before a block is signalled and exception will not be null but the task will not be waited.

Reply:
Ah sorry I am wrong because the background task will throw this is ok I guess then.

Reply:
Yes. We don't want to run the asynchronous task again if we had already thrown an error within it.